File: top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:


    static bool cmp(pair<int, int>& a, pair<int, int>& b)
    {
        return a.second > b.second;
    }
    vector<int> topKFrequent(vector<int>& nums, int k) {
        
        map<int , int> m;

        for(auto it : nums)
        {
            m[it]++;
        }

        vector<pair<int , int>> v;

        for(auto& it : m)
        {
            v.push_back(it);
        }

        sort(v.begin() , v.end(), cmp);

        vector<int> ans;
        // for(auto it : v)
        // {
        //     cout << it.first << " " << it.second << endl;
        // }
        for(int i = k - 1 ; i >= 0 ; i--)
        {
            ans.push_back(v[i].first);
        }

        return ans;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        d = {}

        for i in nums:
            if i not in d:
                d[i] = 1
            else:
                d[i]+=1
        
        l = sorted(d.items() , key = lambda x : x[1] , reverse = True)

        ans = []
        for i in range(k):
            ans.append(l[i][0])
        return ans